### PR TITLE
IR-119: Support IgnoreInvalidImageReferences

### PR DESCRIPTION
**Pull request by dmage**
When the image pruner's spec.ignoreInvalidImageReferences is true, the pruner should be run with --ingore-invalid-refs=true.

This option is enabled by default for new instances.

See also https://github.com/openshift/api/pull/708.

### CODE REVIEW

File: test/e2e/pruner_test.go [hunk 1]
```diff
@@ -242,3 +251,90 @@ func TestPrunerPodCompletes(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	origSpec := cr.Spec.DeepCopy()
+
+	suspend := false
+	cr.Spec.Suspend = &suspend
+	cr.Spec.Schedule = "* * * * *"
+	cr.Spec.IgnoreInvalidImageReferences = true
```
Comment (bparees) [inline test/e2e/pruner_test.go:270]: do you have an inverse test?  (ensures the arg is not passed, or is passed w/ a value of false, when this is not set?)

Comment (dmage) [reply test/e2e/pruner_test.go:284]: now I have, fixed

**Comment by ricardomaraschini** [comment]
@dmage aren't we planning to backport this to release-4.5 ? See https://bugzilla.redhat.com/show_bug.cgi?id=1852501#c39

**Comment by dmage** [comment]
@ricardomaraschini, no, we don't backport API changes

**Pull request merged**